pints/_mcmc/_hamiltonian.py

```diff
@@ -186,11 +212,27 @@ def set_leapfrog_step_size(self, step_size):
         """
         Sets the step size for the leapfrog algorithm.
         """
-        step_size = float(step_size)
-        if step_size <= 0:
+        a = np.atleast_1d(step_size)
+        if len(a[a < 0]) > 0:
             raise ValueError(
-                'Step size for leapfrog algorithm must be greater than zero.')
+                'Step size for leapfrog algorithm must' +
+                'be greater than zero.'
+            )
+        if len(a) == 1:
+            step_size = np.repeat(step_size, self._dimension)
+        elif not len(step_size) == self._dimension:
+            raise ValueError(
+                'Step size should either be of length 1 or equal to the' +
```

Comment:
Allowing for the leapfrog_step_size to be either 1 or of length nparam will also change the hyper-param interface

Reply:
Ok. Can we just assume in the hyper-parameter interface that we are optimising over a vector of length == num_parameters?

Reply:
yea, I think that is fine